### ISSUE TITLE
Change default value of `error_on_failure` to True in network.py

### ISSUE THREAD

**Issue by aptxkid** [box/ClusterRunner#229]
By default, failed network requests in ClusterRunner won't raise unless the caller set `error_on_failure` explicitly. This is generally not pythonic as it's implicit instead of explicit.

**Comment by josephharrington** [comment]
I agree with this in principle but just noting that we'd obviously need to be pretty careful about making this change.
